### PR TITLE
CI Linux: Update platforms

### PR DESCRIPTION
**Pull request by mkoeppe**
### :books: Description

- Adding the newest from `ubuntu`, `linuxmint`, `fedora`, `opensuse`.
- Test a newer 32bit platform.
- Fixing `...-python3.x-minimal` tests broken by #35638.
- A few more parallel jobs because we can.

### :memo: Checklist

- [x] The title is concise, informative, and self-explanatory.
- [x] The description explains in detail what this PR is about.
- [ ] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -350,7 +356,7 @@ setenv =
     #
     # https://hub.docker.com/r/opensuse/leap
     # - OpenSUSE Leap 42 was superseded by the Leap 15 series.
-    # - As of 2022-07, latest = 15 = 15.4
+    # - As of 2023-05, latest = 15 = 15.4
```
Comment (fchapoton) [inline tox.ini]: maybe it should be 15.5 here ?

Comment (dimpase) [reply tox.ini]: indeed. fixed.

**Comment by dimpase** [comment]
I've pushed the fix for linter on #35404

**Review by dimpase — APPROVED** [review]
lgtm

**Comment by mkoeppe** [comment]
Thanks!

**Pull request merged**